### PR TITLE
Make Channel implement Channelable

### PR DESCRIPTION
**Pull request by nachocano**
- Adding `SubscribableTypeStatus` to `ChannelStatus`.
- Marking the Subscriber as Ready. It should actually be read from the PullSubscription status... Maybe in a future PR, or in this one if I find the way.

/cc @n3wscott

### CODE REVIEW

File: pkg/reconciler/channel/channel.go [hunk 1]
```diff
@@ -250,10 +252,11 @@ func (c *Reconciler) syncSubscribers(ctx context.Context, channel *v1alpha1.Chan
 		}
 		c.Recorder.Eventf(channel, corev1.EventTypeNormal, "CreatedSubscriber", "Created Subscriber %q", genName)
 
-		channel.Status.Subscribers = append(channel.Status.Subscribers, eventingduck.SubscriberStatus{
+		channel.Status.SubscribableStatus.Subscribers = append(channel.Status.SubscribableStatus.Subscribers, eventingduck.SubscriberStatus{
 			UID:                s.UID,
 			ObservedGeneration: s.Generation,
-			// TODO: do I need the other fields?
+			// TODO read the condition from PullSubscription, and add a message if it's not ready.
```
Comment (n3wscott) [inline pkg/reconciler/channel/channel.go]: read the condition from PullSubscription, and add a message if it's not ready.

Comment (nachocano) [reply pkg/reconciler/channel/channel.go]: done!

**Pull request merged**